numpyro/infer/elbo.py

```diff
@@ -610,6 +553,38 @@ def get_importance_log_probs(model, guide, args, kwargs, params):
     return model_log_probs, guide_log_probs
 
 
+def _substitute_nonreparam(data, msg):
+    if msg["name"] in data and not msg["fn"].has_rsample:
+        value = msg["fn"](*msg["args"], **msg["kwargs"])
+        value = 0 * value + data[msg["name"]]
+        return value
+
+
+def _get_latents(guide, args, kwargs, params):
+    guide = seed(substitute(guide, data=params), rng_seed=0)
+    guide_tr = trace(guide).get_trace(*args, **kwargs)
```

Comment:
I think unlike `TraceGraph_ELBO` in `TracEnum_ELBO` we need latents from the model trace as well because it can have model enumerated variables.